frontend/test/named-pulls.ts

```diff
@@ -0,0 +1,231 @@
+   pullData({
+      title: "Pending status with no log url",
+      status: {
+         commit_statuses: [status("pending", null)]
+      },
+   }),
+   pullData({
+      title: "One error commit status",
+      status: {
+         commit_statuses: [status("error")]
+      },
+   }),
+   pullData({
+      title: "One successful and one failed commit status",
+      status: {
+         commit_statuses: [status("success"), status("failure")]
+      },
+   }),
+];
+
+export const ManyStatuses = <PullData[]> [
+   pullData({
+      title: "Three commit statuses",
+      status: {
+         commit_statuses: [
+            status("pending"),
+            status("success"),
+            status("error"),
+         ]
+      },
+   }),
+   pullData({
+      title: "Six commit statuses",
+      status: {
+         commit_statuses: [
+            status("error"),
+            status("failure"),
+            status("success"),
+            status("pending"),
+            status("success"),
+            status("error"),
+         ]
+      },
+   }),
+   pullData({
+      title: "Ten commit statuses",
+      status: {
+         commit_statuses: [
+            status("success"),
+            status("error"),
+            status("success"),
+            status("pending"),
+            status("success"),
+            status("failure"),
+            status("success"),
+            status("failure"),
+            status("success"),
+            status("success"),
+         ]
+      },
+   }),
+];
+
+const devBlock = sig(SignatureType.dev_block, true);
+const deployBlock = sig(SignatureType.deploy_block, true);
+export const Blocked = <PullData[]> [
+   pullData({
+      title: "Deploy Blocked",
+      status: {
+         deploy_block: [deployBlock],
+      },
+   }),
+   pullData({
+      title: "Dev Blocked",
+      status: {
+         dev_block: [devBlock],
+      },
+   }),
+];
+
+export const Milestones = <PullData[]> [
+   pullData({
+      title: "With a future Milestone",
+      milestone: {
+         title: "Some big project",
+         due_on: daysAgo(-10),
+      }
+   }),
+   pullData({
+      title: "With a past Milestone",
+      milestone: {
+         title: "Some overdue project",
+         due_on: daysAgo(10),
+      }
+   }),
+];
+
+export const Labels = <PullData[]> [
+   pullData({
+      title: "With a QAing Label applied 1 hour ago",
+      labels: [label({
+         "title": "QAing",
+         "user": "someUser",
+         "created_at": daysAgo(1/24),
+      })],
+   }),
+   pullData({
+      title: "With an external_block Label applied 1 day ago",
+      labels: [label({
+         "title": "external_block",
+         "user": "someOtherUser",
+         "created_at": daysAgo(1),
+      })],
+   }),
+];
+
+export const MyOwn = <PullData[]> [
+   pullData({
+      title: "Pull Created By Me",
+      user: {
+         login: getUser(),
+      }
+   }),
+   pullData({
+      title: "With an up-to-date CR and a QA by Me",
```

Comment:
```suggestion
      title: "With an up-to-date CR and QA by Me",
```

Reply:
I think both are valid. `I ate an apple and an orange`  or  `I ate an apple and orange` both seem sensible with the former sounding more natural to me.

Reply:
Totally. To my ear, at least, though, the parentheses go like
```
With (an up-to-date CR) and (a QA by Me)
```
whereas 
```
With (an up-to-date CR and QA) by Me
```

I can't say for sure, though.